packages/vite/src/node/server/hmr.ts

```diff
@@ -383,6 +385,12 @@ function isNodeWithinCircularImports(
   // It works by checking if any `node` importers are within `nodeChain`, which
   // means there's an import loop with a HMR-accepted module in it.
 
+  if (traversedModules.has(node)) {
+    // To avoid infinite recursion, we only check each module once.
```

Comment:
I'd probably remove this for now since we're not sure how it's triggered. The `currentChain` check should already avoid the infinite recursion
```suggestion
```